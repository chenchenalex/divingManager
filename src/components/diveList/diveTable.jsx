import React from "react";
import Table, {
  TableBody,
  TableCell,
  TableHead,
  TableRow
} from "material-ui/Table";
import { Link } from "react-router-dom";

const DivingTable = ({ savedData }) => {
  return (
    <Table className="testing">
      <TableHead>
        <TableRow>
          <TableCell>Name</TableCell>
          <TableCell>Location</TableCell>
          <TableCell>Date</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {savedData.map(dive => {
          return (
            <TableRow key={dive.id}>
              <TableCell>
                <Link to={`/list/edit/${dive.id}`}>{dive.name}</Link>
              </TableCell>
              <TableCell>{dive.location}</TableCell>
              <TableCell>{dive.date}</TableCell>
            </TableRow>
          );
        })}
      </TableBody>
    </Table>
  );
};

export default DivingTable;
