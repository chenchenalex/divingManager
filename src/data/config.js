export const INITIAL_FORM_DATA = {
  name: "",
  country: "",
  date: "2018-01-01",
  depth: "",
  oxygen: ""
};

export const FORM_CONFIG = {
  name: {
    required: true,
    invalid: false,
    label: "Name",
    helperText: "Please provide name here"
  },
  country: {
    required: true,
    invalid: false,
    label: "Country",
    helperText: "Please select a country"
  },
  date: {
    required: true,
    invalid: false,
    label: "Date",
    helperText: "Date is missing"
  },
  depth: {
    required: true,
    invalid: false,
    label: "Depth",
    helperText: "Depth is missing"
  },
  oxygen: {
    required: false,
    invalid: false,
    label: "Oxygen left",
    helperText: "Oxygen amount is not provided"
  },
  feedback: {
    multiline: true,
    rowsMax: "6",
    required: false,
    invalid: false,
    label: "How do you feel?",
    helperText: "No feedback?"
  }
};
