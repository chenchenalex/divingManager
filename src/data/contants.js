export const INITIAL_FORM_DATA = {
  name: "",
  location: "",
  date: "2018-01-01"
};
