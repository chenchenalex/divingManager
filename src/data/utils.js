const getDives = ({ diveById }) => {
  if (typeof diveById === "undefined") return [];

  return Object.keys(diveById).map(id => diveById[id]);
};

export default getDives;
