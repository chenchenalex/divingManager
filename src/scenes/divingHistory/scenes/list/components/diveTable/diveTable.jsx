import React from "react";
import Table, {
  TableBody,
  TableCell,
  TableHead,
  TableRow
} from "material-ui/Table";
import Checkbox from "material-ui/Checkbox";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const DivingTable = ({ state, tableData, onSelect, onSelectAll }) => {
  const checkSelected = ({ id }) => {
    return state.selected.includes(id);
  };

  return (
    <Table className="testing">
      <TableHead>
        <TableRow>
          <TableCell>
            <Checkbox
              onChange={onSelectAll}
              indeterminate={
                state.selected.length > 0 &&
                state.selected.length < tableData.length
              }
              checked={
                state.selected.length > 0 &&
                state.selected.length === tableData.length
              }
            />
          </TableCell>
          <TableCell>Name</TableCell>
          <TableCell>Location</TableCell>
          <TableCell>Date</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {tableData.map(dive => {
          const isSelected = checkSelected(dive);
          return (
            <TableRow
              key={dive.id}
              role="checkbox"
              aria-checked={isSelected}
              tabIndex={-1}
              selected={isSelected}
            >
              <TableCell>
                <Checkbox
                  onChange={onSelect}
                  value={dive.id}
                  checked={isSelected}
                />
              </TableCell>
              <TableCell>
                <Link to={`/list/edit/${dive.id}`}>{dive.name}</Link>
              </TableCell>
              <TableCell>{dive.location}</TableCell>
              <TableCell>{dive.date}</TableCell>
            </TableRow>
          );
        })}
      </TableBody>
    </Table>
  );
};

DivingTable.proptypes = {
  state: PropTypes.object.isRequired,
  tableData: PropTypes.array.isRequired,
  onSelect: PropTypes.func.isRequired,
  onSelectAll: PropTypes.func.isRequired
};

export default DivingTable;
