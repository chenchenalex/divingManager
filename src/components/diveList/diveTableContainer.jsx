import React from "react";
import { connect } from "react-redux";
import DiveTable from "./diveTable";
import DivingListContainer from "./style";
import AddButton from "./addNew";
import { getDives } from "../../data/utils";

class divingListComponent extends React.Component {
  render() {
    return (
      <DivingListContainer className="diving-list">
        <AddButton />
        <DiveTable savedData={this.props.divingHistory} />
      </DivingListContainer>
    );
  }
}

function mapStateToProps(state) {
  return {
    divingHistory: getDives(state.divingHistory)
  };
}

export default connect(mapStateToProps)(divingListComponent);
