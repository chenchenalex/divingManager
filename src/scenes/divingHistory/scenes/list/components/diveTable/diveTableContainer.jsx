import React, { Fragment } from "react";
import store from "src/store";
import { deleteActionCreator, userFetchDataAsync } from "../../actions";
import { connect } from "react-redux";

// components
import DiveTable from "./diveTable";
import { DivingContainer } from "../../style";
import AddButton from "../buttons/addNew";
import DeleteButton from "../buttons/delete";
import bgImg from "src/assets/images/dive_bg1.jpg";
import ImageBanner from "src/components/imageBanner";

// utils
import { getDives } from "src/data/utils";

const { dispatch } = store;
export class DivingListComponent extends React.Component {
  state = {
    selected: []
  };

  componentDidMount() {
    dispatch(userFetchDataAsync("alex"));
  }

  onSelectAll = event => {
    const isChecked = event.target.checked;

    if (isChecked) {
      this.setState(state => ({
        selected: this.props.divingHistory.map(dive => dive.id)
      }));
    } else {
      this.setState(state => ({
        selected: []
      }));
    }
  };

  onSelect = event => {
    const isChecked = event.target.checked;
    const id = event.target.value;

    if (isChecked) {
      this.setState(state => {
        return {
          selected: [...state.selected, id]
        };
      });
    } else {
      this.setState(state => {
        return {
          selected: [...state.selected.filter(selectedId => selectedId !== id)]
        };
      });
    }
  };

  onDelete = () => {
    dispatch(deleteActionCreator(this.state.selected));

    this.setState(state => {
      return {
        selected: []
      };
    });
  };

  render() {
    return (
      <Fragment>
        <ImageBanner
          imgSrc={bgImg}
          imgAlt="diver hero image"
          title="DIVE HISTORY"
        />
        <DivingContainer className="diving-list">
          <AddButton />
          <DeleteButton onDelete={this.onDelete} state={this.state} />
          <DiveTable
            state={this.state}
            tableData={this.props.divingHistory || []}
            onSelect={this.onSelect}
            onSelectAll={this.onSelectAll}
          />
        </DivingContainer>
      </Fragment>
    );
  }
}

function mapStateToProps(state) {
  return {
    divingHistory: getDives(state.scenes.divingHistory)
  };
}

export default connect(mapStateToProps)(DivingListComponent);
