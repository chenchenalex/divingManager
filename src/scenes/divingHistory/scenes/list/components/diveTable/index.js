import diveTableContainer from "./diveTableContainer";

export default diveTableContainer;
