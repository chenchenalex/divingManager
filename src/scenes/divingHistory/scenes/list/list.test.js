import { DivingListComponent } from "./components/diveTable/diveTableContainer";
import AddNew from "./components/buttons/addNew";
import DeleteBtn from "./components/buttons/delete";
import DiveTable from "./components/diveTable/diveTable";
import { prevState as mockState } from "../../mock.data";
import { getDives } from "src/data/utils";
import { shallow } from "enzyme";

import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

const divingHistory = getDives({ diveById: mockState });

describe("Dive list: dive table container tests", () => {
  const container = shallow(
    <DivingListComponent divingHistory={divingHistory} />
  );

  it("onSelectAll: should select all dives or deselect them all", () => {
    const instance = container.instance();
    const event1 = {
      target: {
        checked: true
      }
    };

    const event2 = {
      target: {
        checked: false
      }
    };

    const selectedDives = ["abc", "def"];

    instance.onSelectAll(event1);
    expect(instance.state.selected).toEqual(selectedDives);

    instance.onSelectAll(event2);
    expect(instance.state.selected).toEqual([]);
  });

  it("onSelect: should select dive by id", () => {
    const instance = container.instance();
    const event1 = {
      target: {
        checked: true,
        value: "abc"
      }
    };

    const event2 = {
      target: {
        checked: false,
        value: "abc"
      }
    };
    // select dive

    instance.onSelect(event1);
    expect(instance.state.selected).toContain("abc");

    instance.onSelect(event2);
    expect(instance.state.selected).not.toContain("abc");
  });

  it("onDelete: should delete dive by id", () => {
    const instance = container.instance();
    const event1 = {
      target: {
        checked: true,
        value: "abc"
      }
    };

    instance.onSelect(event1);
    instance.onDelete();

    expect(instance.state.selected).toEqual([]);
  });
});

describe("Dive list: Snapshots tests", () => {
  it("addNew: should render correctly", () => {
    const addNewButton = renderer.create(
      <MemoryRouter>
        <AddNew />
      </MemoryRouter>
    );

    expect(addNewButton).toMatchSnapshot();
  });

  it("delete button: should render correctly", () => {
    const onDelete = jest.fn();
    const state = { selected: [] };

    const deleteBtn = renderer.create(
      <MemoryRouter>
        <DeleteBtn onDelete={onDelete} state={state} />
      </MemoryRouter>
    );

    expect(deleteBtn).toMatchSnapshot();
  });

  it("Dive table: should render correctly", () => {
    const onSelect = jest.fn();
    const onSelectAll = jest.fn();
    const state = { selected: [] };

    const diveTable = renderer.create(
      <MemoryRouter>
        <DiveTable
          state={state}
          tableData={divingHistory}
          onSelect={onSelect}
          onSelectAll={onSelectAll}
        />
      </MemoryRouter>
    );

    expect(diveTable).toMatchSnapshot();
  });
});
