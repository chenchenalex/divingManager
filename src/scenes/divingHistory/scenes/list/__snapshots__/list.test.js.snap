// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dive list: Snapshots tests Dive table: should render correctly 1`] = `
<table
  className="MuiTable-root-26 testing sc-bwzfXH fGCCLo"
>
  <thead
    className="MuiTableHead-root-27"
  >
    <tr
      className="MuiTableRow-root-28 MuiTableRow-head-31"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-head-34"
        scope="col"
      >
        <span
          className="MuiButtonBase-root-16 MuiIconButton-root-50 MuiSwitchBase-root-46 MuiCheckbox-root-41 MuiCheckbox-colorSecondary-45"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-55"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-56"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              checked={false}
              className="MuiSwitchBase-input-49"
              onChange={[Function]}
              type="checkbox"
            />
          </span>
          <span
            className="MuiTouchRipple-root-19"
          />
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-head-34"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-head-34"
        scope="col"
      >
        Country
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-head-34"
        scope="col"
      >
        Date
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-62"
  >
    <tr
      aria-checked={false}
      className="MuiTableRow-root-28"
      role="checkbox"
      tabIndex={-1}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        <span
          className="MuiButtonBase-root-16 MuiIconButton-root-50 MuiSwitchBase-root-46 MuiCheckbox-root-41 MuiCheckbox-colorSecondary-45"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-55"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-56"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              checked={false}
              className="MuiSwitchBase-input-49"
              onChange={[Function]}
              type="checkbox"
              value="test1"
            />
          </span>
          <span
            className="MuiTouchRipple-root-19"
          />
        </span>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        <a
          href="/list/edit/test1"
          onClick={[Function]}
        >
          kaotao OW/AOW lesson
        </a>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        Thailand
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        2017-03-15
      </td>
    </tr>
    <tr
      aria-checked={false}
      className="MuiTableRow-root-28"
      role="checkbox"
      tabIndex={-1}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        <span
          className="MuiButtonBase-root-16 MuiIconButton-root-50 MuiSwitchBase-root-46 MuiCheckbox-root-41 MuiCheckbox-colorSecondary-45"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-55"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-56"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              checked={false}
              className="MuiSwitchBase-input-49"
              onChange={[Function]}
              type="checkbox"
              value="test2"
            />
          </span>
          <span
            className="MuiTouchRipple-root-19"
          />
        </span>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        <a
          href="/list/edit/test2"
          onClick={[Function]}
        >
          Philipine diving with Eric
        </a>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        Philipine
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        2018-03-15
      </td>
    </tr>
    <tr
      aria-checked={false}
      className="MuiTableRow-root-28"
      role="checkbox"
      tabIndex={-1}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        <span
          className="MuiButtonBase-root-16 MuiIconButton-root-50 MuiSwitchBase-root-46 MuiCheckbox-root-41 MuiCheckbox-colorSecondary-45"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-55"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-56"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              checked={false}
              className="MuiSwitchBase-input-49"
              onChange={[Function]}
              type="checkbox"
              value="test3"
            />
          </span>
          <span
            className="MuiTouchRipple-root-19"
          />
        </span>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        <a
          href="/list/edit/test3"
          onClick={[Function]}
        >
          Free diving with sharks
        </a>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        Cannes
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        2019-05-25
      </td>
    </tr>
    <tr
      aria-checked={false}
      className="MuiTableRow-root-28"
      role="checkbox"
      tabIndex={-1}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        <span
          className="MuiButtonBase-root-16 MuiIconButton-root-50 MuiSwitchBase-root-46 MuiCheckbox-root-41 MuiCheckbox-colorSecondary-45"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-55"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-56"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              checked={false}
              className="MuiSwitchBase-input-49"
              onChange={[Function]}
              type="checkbox"
              value="test4"
            />
          </span>
          <span
            className="MuiTouchRipple-root-19"
          />
        </span>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        <a
          href="/list/edit/test4"
          onClick={[Function]}
        >
          Blue hold diving, first CAVE DIVE!
        </a>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        Mexico
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-33 MuiTableCell-body-35"
      >
        2020-03-25
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Dive list: Snapshots tests addNew: should render correctly 1`] = `
<a
  href="/list/add"
  onClick={[Function]}
>
  <button
    aria-label="add"
    className="MuiButtonBase-root-16 MuiButton-root-1 MuiButton-raised-6 MuiButton-fab-11 MuiButton-raisedPrimary-7"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </g>
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-19"
    />
  </button>
</a>
`;

exports[`Dive list: Snapshots tests delete button: should render correctly 1`] = `
<button
  aria-label="delete"
  className="MuiButtonBase-root-16 MuiButtonBase-disabled-17 MuiButton-root-1 MuiButton-raised-6 MuiButton-fab-11 MuiButton-disabled-10"
  disabled={true}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </g>
    </svg>
  </span>
</button>
`;
